OfficeDevPnP.Core/OfficeDevPnP.Core/AppModelExtensions/FieldAndContentTypeExtensions.cs

```diff
@@ -592,10 +647,17 @@ public static void AddContentTypeToListByName(this List list, string contentType
         public static void AddContentTypeToList(this List list, ContentType contentType, bool defaultContent = false)
         {
             if (contentType == null)
+            {
                 throw new ArgumentNullException("contentType");
+            }
 
             if (list.ContentTypeExistsById(contentType.Id.StringValue))
+            {
                 return;
+            }
+
+            list.Context.Load(list, l => l.ContentTypesEnabled);
```

Comment:
What about check if the property is available first before loading it instead of assuming it's not and always loading it? E.g. 
```
if(!list.IsPropertyAvailable("ContentTypesEnabled") 
{ 
// load the property 
}
```